octoprint_octobullet/__init__.py

```diff
@@ -29,10 +33,106 @@ def __init__(self):
 		self._bullet = None
 		self._channel = None
 		self._sender = None
+		self._pc_interval = 5
+		self._message_count = 0
+		self._quiet_time_sec = 1800
+		self._last_message = 0
+		self._etl_format   = "{days:d}:{hours:02d}h {minutes:02d}m"
+		self._eta_strftime = "%H:%M %d-%m"
 
 	def _connect_bullet(self, apikey, channel_name=""):
 		self._bullet, self._sender = self._create_sender(apikey, channel_name)
 
+	#~~ progress message helpers
+	def _sanitize_current_data(self, currentData):
```

Comment:
I think it's dangerous to just use arbitrary values (1000, 0, ...) instead of the actual values. If they are `None` they are `None` for a reason - they are unknown or not stable and shouldn't be displayed to the user.